### DISCUSSION TITLE
paused DAGs still generate (large) logs

### DISCUSSION THREAD

**Discussion by atpage** [apache/airflow#23363] [General]
### Apache Airflow version

2.2.2

### What happened

I noticed that my log folder (`~/airflow/logs/`) was getting very large.  To my surprise, 90% of it is from the example DAGs, which are all paused.  These disabled examples are generating ~110MB of logs every day.  I can't find a solution other than (presumably) removing the examples.

### What you think should happen instead

Paused DAGs shouldn't be continuously generating log entries.  Or maybe there should be a setting to disable logging for disabled DAGs?

### Operating System

Ubuntu 21.10

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Converting that to a discussion. You have not posted what kind of logs you are seing. Please add information, names, which are the big ones and copy parts of those so that it can be reasoned about.  We do not know how large is large and what is causing that - it migth be simply bugs in your DAGs.

Getting rid of the logs  might or might not be possible -depending on the kind of logs you have.  And there are also  other ways that can actually help you.

Generally all DAGs are parsed by scheduler regardless if they are paused - this is by design and is not going to change. If parsing of the dags produces a lot of logs - you shoudl make sure they are not produced. Tthis is entirely expected to get the DAGs parsed continuously even if parsed. 

We want to have DAGs refreshed with the latest structure regardless if they are paused or not so that you can unpause them quickly.

Normally DAGs should not generate a lot of logs unless your example dags are doing that on purpose during parsing. so if they are generating a lot of logs, this likely the problem of your DAGs that need fixing.

And as a last resort you can always add .airflowignore if you want to disable the dags from being parsed.

**Reply by atpage** [reply]
I'm referring to the example DAGs from airflow itself.  Not anything that I made.

```
$ tree -h ~/airflow/logs/scheduler/2022-04-29/native_dags/example_dags/

~/airflow/logs/scheduler/2022-04-29/native_dags/example_dags/
├── [680K]  example_bash_operator.py.log
├── [725K]  example_branch_datetime_operator.py.log
├── [730K]  example_branch_day_of_week_operator.py.log
├── [680K]  example_branch_labels.py.log
├── [687K]  example_branch_operator.py.log
├── [731K]  example_branch_python_dop_operator_3.py.log
├── [645K]  example_complex.py.log
├── [668K]  example_dag_decorator.py.log
├── [830K]  example_external_task_marker_dag.py.log
├── [788K]  example_kubernetes_executor_config.py.log
├── [691K]  example_kubernetes_executor.py.log
├── [672K]  example_latest_only.py.log
├── [722K]  example_latest_only_with_trigger.py.log
├── [698K]  example_nested_branch_dag.py.log
├── [755K]  example_passing_params_via_test_command.py.log
├── [676K]  example_python_operator.py.log
├── [718K]  example_short_circuit_operator.py.log
├── [670K]  example_skip_dag.py.log
├── [1.1M]  example_subdag_operator.py.log
├── [711K]  example_task_group_decorator.py.log
├── [672K]  example_task_group.py.log
├── [707K]  example_time_delta_sensor_async.py.log
├── [715K]  example_trigger_controller_dag.py.log
├── [693K]  example_trigger_target_dag.py.log
├── [763K]  example_xcomargs.py.log
├── [645K]  example_xcom.py.log
├── [4.0K]  plugins
│   └── [486K]  workday.py.log
├── [4.0K]  subdags
│   └── [483K]  subdag.py.log
├── [630K]  test_utils.py.log
├── [648K]  tutorial_etl_dag.py.log
├── [633K]  tutorial.py.log
├── [683K]  tutorial_taskflow_api_etl.py.log
└── [726K]  tutorial_taskflow_api_etl_virtualenv.py.log

2 directories, 33 files
```

That's 20+ MB from just the last 4-5 hours.  Here's a snippet of the first file.  This repeats about twice a minute:

```
[2022-04-29 08:36:12,435] {processor.py:163} INFO - Started process (PID=2057861) to work on /usr/local/lib/python3.9/dist-packages/airflow/example_dags/example_bash_operator.py
[2022-04-29 08:36:12,435] {processor.py:625} INFO - Processing file /usr/local/lib/python3.9/dist-packages/airflow/example_dags/example_bash_operator.py for tasks to queue
[2022-04-29 08:36:12,436] {logging_mixin.py:109} INFO - [2022-04-29 08:36:12,436] {dagbag.py:500} INFO - Filling up the DagBag from /usr/local/lib/python3.9/dist-packages/airflow/example_dags/example_bash_operator.py
[2022-04-29 08:36:12,460] {processor.py:637} INFO - DAG(s) dict_keys(['example_bash_operator']) retrieved from /usr/local/lib/python3.9/dist-packages/airflow/example_dags/example_bash_operator.py
[2022-04-29 08:36:12,466] {logging_mixin.py:109} INFO - [2022-04-29 08:36:12,466] {dag.py:2396} INFO - Sync 1 DAGs
[2022-04-29 08:36:12,479] {logging_mixin.py:109} INFO - [2022-04-29 08:36:12,479] {dag.py:2928} INFO - Setting next_dagrun for example_bash_operator to 2022-04-28 00:00:00+00:00
[2022-04-29 08:36:12,488] {processor.py:171} INFO - Processing /usr/local/lib/python3.9/dist-packages/airflow/example_dags/example_bash_operator.py took 0.055 seconds
```

> If parsing of the dags produces a lot of logs - you shoudl make sure they are not produced.

So I guess my question is, how can I do that?  Is there a way to turn down the logging level for certain DAGs?

For now I just set a cron job to delete them all daily.

**Reply by potiuk** [reply]
Sure. Disable them. If this is about airflow example dags - there is no reason to have them in your prod environment at all. They should be disabled and this is default setting. Just look in airflow configuration and set "|example dags" to false.

**Reply by potiuk** [reply]
https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#load-examples

> Whether to load the DAG examples that ship with Airflow. It's good to get started, but you probably want to set this to False in a production environment

**Reply by atpage** [reply]
So the examples should really *never* be loaded, except in a short-lived dev environment, because otherwise they'll slowly fill the hard drive, even while paused?  But that's not a bug?  :confused: 

I guess my cron job fixes it for now.  I like having the examples available for reference.

**Reply by potiuk** [reply]
No. AND dags will slowly fill the logs. This is not a bug. This is a feature. 

You are supposed to manage logs on your own. Airflow (like most modern software) produces logs and let the deployment remove and archive the logs. Pretty much every other software that you install in production deployment does it. 

So your cron job is exactly what it should be.

**Reply by noobjun** [reply]
I ran into this issue as well, it's very misleading for PAUSED dags to still generate logs, there are nearly 30 example DAGs, generating ~110M everyday. Today I found my airflow box non-responsive due to out of space caused by this.

Surely you can say do not put these in prod, but even in dev environment, it is not expected for them to fill up the hard drive while on pause.

**Reply by tanelk** [reply]
It is not DAG logs per se, but it is file parsing logs. The DAG parser "executes" all files that could contain DAG definitions at regular interval and sees what DAGs it finds. Yes, during last parsing there was a DAG, that is paused. But it is python code - it is dynamic, next time there could be a whole new DAG there. So we must parse it and it will produce logs. If you do not want logs, then you could regularly  delete them using a preferred tool. Or you could add files that can not contain DAGs to .airflowignore.
